plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/IsNullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,209 @@
+        }
+
+        try (TemporaryRelation temporaryRelation = dataSetup.setupTemporaryRelation(unmodifiableList(testCases))) {
+            verifySelect(queryRunner, session, temporaryRelation);
+            verifyPredicate(queryRunner, session, temporaryRelation, IS_NULL, false, List.of(TUPLE_DOMAIN_ONLY, ALWAYS).contains(pushdownImplementation));
+            verifyPredicate(queryRunner, session, temporaryRelation, IS_NULL, true, List.of(CONNECTOR_EXPRESSION_ONLY, ALWAYS).contains(pushdownImplementation));
+            verifyPredicate(queryRunner, session, temporaryRelation, IS_NOT_NULL, false, List.of(TUPLE_DOMAIN_ONLY, ALWAYS).contains(pushdownImplementation));
+            verifyPredicate(queryRunner, session, temporaryRelation, IS_NOT_NULL, true, List.of(CONNECTOR_EXPRESSION_ONLY, ALWAYS).contains(pushdownImplementation));
+        }
+        return this;
+    }
+
+    private void verifySelect(QueryRunner queryRunner, Session session, TemporaryRelation temporaryRelation)
+    {
+        // Closing QueryAssertions would close the QueryRunner
+        QueryAssertions queryAssertions = new QueryAssertions(queryRunner);
+
+        QueryAssertions.ResultAssert actual = assertThat(queryAssertions.query(session, "SELECT * FROM " + temporaryRelation.getName()))
+                .result();
+        MaterializedResult expected = queryRunner.execute(session, testCases.stream()
+                .map(TestCase::expectedLiteral)
+                .collect(joining(",", "VALUES ROW(", ")")));
+
+        // Verify types if specified
+        for (int column = 0; column < testCases.size(); column++) {
+            TestCase testCase = testCases.get(column);
+            actual.hasType(column, testCase.expectedType());
+            assertThat(expected.getTypes())
+                    .as(format("Expected literal type at column %d (check consistency of expected type and expected literal)", column + 1))
+                    .element(column)
+                    .isEqualTo(testCase.expectedType());
+        }
+
+        actual.matches(expected);
+    }
+
+    private void verifyPredicate(QueryRunner queryRunner, Session session, TemporaryRelation temporaryRelation, Scenario scenario, boolean connectorExpression, boolean expectPushdown)
+    {
+        TestCase firstCase = testCases.getFirst();
+        String firstColumnName = "col_0";
+        String withConnectorExpression = connectorExpression ? " OR %s IS NULL".formatted(firstColumnName) : "";
+
+        String queryWithAll = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " +
+                IntStream.range(SPECIAL_COLUMNS, testCases.size())
+                        .mapToObj(column -> getPredicate(column, scenario))
+                        .collect(joining(" AND "))
+                + withConnectorExpression;
+
+        // Closing QueryAssertions would close the QueryRunner
+        QueryAssertions queryAssertions = new QueryAssertions(queryRunner);
```

Comment:
Can we inject `queryAssertions` instead of query runner ?

Reply:
Also not sure if it gives benefits.